src/main/java/com/google/sps/servlets/RecipePostServlet.java

```diff
@@ -65,12 +66,40 @@
 
   // Name of the index used.
   private static final String INDEX_NAME = "recipes_index";
+
+  /**
+   * doGet retrieves the attributes of the recipe to be edited and sets them as reqest attributes
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity recipeEntity = null;
+    try {
+      recipeEntity = getRecipeById(datastore, request.getParameter("id"));
+    } catch (EntityNotFoundException e) {
+      request.setAttribute("error", 1);
```

Comment:
you can do

request.setAttribute("error", 1);
request.getRequestDispatcher("/recipe_post.jsp").forward(request, response);
return;

And remove the

if (recipeEntity != null) {

below.